### PR TITLE
[SPARK-19386][SPARKR][DOC] Bisecting k-means in SparkR documentation

### PR DESCRIPTION
**Pull request by krishnakalyan3**
## What changes were proposed in this pull request?
Update programming guide, example and vignette with Bisecting k-means.

### CODE REVIEW

File: R/pkg/vignettes/sparkr-vignettes.Rmd [hunk 1]
```diff
@@ -494,6 +494,8 @@ SparkR supports the following machine learning models and algorithms.
 
 * Latent Dirichlet Allocation (LDA)
 
+* Bisecting $k$-means
```
Comment (felixcheung) [inline R/pkg/vignettes/sparkr-vignettes.Rmd]: please sort this

File: R/pkg/vignettes/sparkr-vignettes.Rmd [hunk 2]
```diff
@@ -819,6 +821,18 @@ perplexity <- spark.perplexity(model, corpusDF)
 perplexity
 ```
 
+#### Bisecting k-means
```
Comment (felixcheung) [inline R/pkg/vignettes/sparkr-vignettes.Rmd]: ditto for this

**Comment by felixcheung** [comment]
@wangmiao1981

File: docs/ml-clustering.md [hunk 3]
```diff
@@ -167,6 +167,13 @@ Refer to the [Python API docs](api/python/pyspark.ml.html#pyspark.ml.clustering.
 
 {% include_example python/ml/bisecting_k_means_example.py %}
 </div>
+
+<div data-lang="r" markdown="1">
+
+Refer to the [R API docs](api/R/spark.bisectingKmeans.html) for more details. {% include_example r/ml/bisectingKmeans.R %}
```
Comment (felixcheung) [inline docs/ml-clustering.md]: you will need to add this file r/ml/bisectingKmeans.R

Comment (felixcheung) [inline docs/ml-clustering.md]: and this ` {% include_example r/ml/bisectingKmeans.R %}` should be a separate line

File: docs/ml-clustering.md [hunk 4]
```diff
@@ -167,6 +167,13 @@ Refer to the [Python API docs](api/python/pyspark.ml.html#pyspark.ml.clustering.
...
+
+{% include_example r/ml/lda.R %}
```
Comment (felixcheung) [inline docs/ml-clustering.md]: this is incorrect

File: R/pkg/vignettes/sparkr-vignettes.Rmd [hunk 5]
```diff
@@ -84,7 +84,7 @@ class(carsGP)
 
 SparkR supports a number of commonly used machine learning algorithms. Under the hood, SparkR uses MLlib to train the model. Users can call `summary` to print a summary of the fitted model, `predict` to make predictions on new data, and `write.ml`/`read.ml` to save/load fitted models.
 
-SparkR supports a subset of R formula operators for model fitting, including ‘~’, ‘.’, ‘:’, ‘+’, and ‘-‘. We use linear regression as an example.
+SparkR supports a subset of R formula operators for model fitting, including ???~???, ???.???, ???:???, ???+???, and ???-???. We use linear regression as an example.
```
Comment (wangmiao1981) [inline R/pkg/vignettes/sparkr-vignettes.Rmd]: why ???

File: R/pkg/vignettes/sparkr-vignettes.Rmd [hunk 6]
```diff
@@ -373,7 +373,7 @@ head(out, 3)
...
+`gapply` can apply a function to each group of a `SparkDataFrame`. The function is to be applied to each group of the `SparkDataFrame` and should have only two parameters: grouping key and R `data.frame` corresponding to that key. The groups are chosen from `SparkDataFrames` column(s). The output of function should be a `data.frame`. Schema specifies the row format of the resulting `SparkDataFrame`. It must represent R function???s output schema on the basis of Spark data types. The column names of the returned `data.frame` are set by user. See [here](#DataTypes) for mapping between R and Spark.
```
Comment (wangmiao1981) [inline R/pkg/vignettes/sparkr-vignettes.Rmd]: I don't see any change here.

File: R/pkg/vignettes/sparkr-vignettes.Rmd [hunk 7]
```diff
@@ -561,7 +563,7 @@ summary(model)
 
 #### Multilayer Perceptron
 
-Multilayer perceptron classifier (MLPC) is a classifier based on the [feedforward artificial neural network](https://en.wikipedia.org/wiki/Feedforward_neural_network). MLPC consists of multiple layers of nodes. Each layer is fully connected to the next layer in the network. Nodes in the input layer represent the input data. All other nodes map inputs to outputs by a linear combination of the inputs with the node’s weights $w$ and bias $b$ and applying an activation function. This can be written in matrix form for MLPC with $K+1$ layers as follows:
+Multilayer perceptron classifier (MLPC) is a classifier based on the [feedforward artificial neural network](https://en.wikipedia.org/wiki/Feedforward_neural_network). MLPC consists of multiple layers of nodes. Each layer is fully connected to the next layer in the network. Nodes in the input layer represent the input data. All other nodes map inputs to outputs by a linear combination of the inputs with the node???s weights $w$ and bias $b$ and applying an activation function. This can be written in matrix form for MLPC with $K+1$ layers as follows:
```
Comment (wangmiao1981) [inline R/pkg/vignettes/sparkr-vignettes.Rmd]: I don't see the change here either.

File: examples/src/main/r/ml/bisectingKmeans.R [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+# ./bin/spark-submit examples/src/main/r/ml/bisectingKmeans.R
+
+# Load SparkR library into your R session
+library(SparkR)
+
+# Initialize SparkSession
+sparkR.session(appName = "SparkR-ML-bisectingKmeans-example")
+
+# $example on$
+irisDF <- suppressWarnings(createDataFrame(iris))
```
Comment (wangmiao1981) [inline examples/src/main/r/ml/bisectingKmeans.R]: `suppressWarnings` should not appear in example.

File: R/pkg/vignettes/sparkr-vignettes.Rmd (hunk 1 above)
Comment (krishnakalyan3) [inline R/pkg/vignettes/sparkr-vignettes.Rmd]: @felixcheung could you please let me know whats wrong here?

File: examples/src/main/r/ml/bisectingKmeans.R [hunk 9]
```diff
@@ -0,0 +1,42 @@
...
+
+# Fit bisecting k-means model with four centers
+model <- spark.bisectingKmeans(df, Sepal_Length ~ Sepal_Width, k = 4)
+
+# get fitted result from a bisecting k-means model
+fitted.model <- fitted(model, "centers")
+
+# Model summary
+showDF(fitted.model)
```
Comment (wangmiao1981) [inline examples/src/main/r/ml/bisectingKmeans.R]: Model Summary should use `summary` method. Otherwise, change the comment.

**Comment by krishnakalyan3** [comment]
@wangmiao1981 sorry, had made a erroneous commit. Could you please review the PR?.

File: R/pkg/vignettes/sparkr-vignettes.Rmd (hunk 1 above)
Comment (felixcheung) [inline R/pkg/vignettes/sparkr-vignettes.Rmd]: these model names are in order
```
#### Clustering
 
 * Gaussian Mixture Model (GMM)
 
 * $k$-means Clustering
  
  * Latent Dirichlet Allocation (LDA)
  
 * Bisecting $k$-means
```

should be
```
#### Clustering
 
 * Bisecting $k$-means

 * Gaussian Mixture Model (GMM)
 
 * $k$-means Clustering
  
  * Latent Dirichlet Allocation (LDA)
  
 ```

File: R/pkg/vignettes/sparkr-vignettes.Rmd (hunk 2 above)
Comment (felixcheung) [inline R/pkg/vignettes/sparkr-vignettes.Rmd]: same here. the model sections are in alphabetic order

**Comment by krishnakalyan3** [comment]
ping @wangmiao1981 @felixcheung

**Comment by felixcheung** [comment]
looking good to me. I'll wait for @wangmiao1981 to review.

**Comment by wangmiao1981** [comment]
LGTM

**Comment by felixcheung** [comment]
merged to master

**Comment by wangmiao1981** [comment]
Jenkins build was not triggered for the last revision. It breaks the build.

**Comment by felixcheung** [comment]
ouch, Jenkins was never run the entire time. I need to go double check why AppVeyor is passing.

**Pull request closed without merging**